src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const arr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const prop of action.keysToRemove) {
+          if (clone.hasOwnProperty(prop)) {
+            delete clone[prop];
+          }
+        }
+        break;
+
+      default:
+        for (const key in clone) {
+          delete clone[key];
+        }
```

Comment:
i believe it would be okay to make clone a non-constant variable and just do here clone = {}